docs/cheat-sheets/ruby-command-injection.mdx

```diff
@@ -88,17 +90,227 @@ open(user_input)
+- Always try to use internal Ruby API (if it exists) instead of running an OS command. Use internal language features instead of invoking commands that can be exploited.
+- If the use of user input is unavoidable, create an allowlist for inputs, such as allowed arguments.
+- Strip everything except alphanumeric characters from an input provided for the command string and arguments.
+
+#### Semgrep rule
+
+<LinkToRegistryRule ruleId="ruby.lang.security.dangerous-exec.dangerous-exec" />
+
+### 1.D. exec() function
+
+The `exec()` executes OS commands. This might potentially lead to command injection vulnerability when used with user input. The malicious actor can potentially run OS commands to exploit the system.
+
+Example:
+
+```ruby
+# safe
+exec("ls -lah /tmp")
+
+# vulnerable
+user_input = ' && cat /etc/passwd' # Value supplied by user
+exec("ls #{user_input}")
+```
+
+#### References
+
+- [`exec()` documentation](https://apidock.com/ruby/Kernel/exec)
+
+#### Mitigation
+
+- Do not provide raw user input to `open` methods.
+- Always try to use internal Ruby API (if it exists) instead of running an OS command. Use internal language features instead of invoking commands that can be exploited.
+- If the use of user input is unavoidable, create an allowlist for inputs, such as allowed arguments.
+- Strip everything except alphanumeric characters from an input provided for the command string and arguments.
+
+#### Semgrep rule
+
+<LinkToRegistryRule ruleId="ruby.lang.security.dangerous-exec.dangerous-exec" />
+
+### 1.D. spawn() function
+
+The `spawn()` function executes OS commands. This might potentially lead to command injection vulnerability when used with user input. The malicious actor can potentially run OS commands to exploit the system.
+
+Example:
+
+```ruby
+# safe
+pid = spawn("ls -lah /tmp")
+Process.wait pid
+
+# vulnerable
+user_input = ' && cat /etc/passwd' # Value supplied by user
+pid = spawn("ls #{user_input}")
+Process.wait pid
+```
+
+#### References
+
+- [`spawn()` documentation](https://apidock.com/ruby/Kernel/spawn)
+
+#### Mitigation
+
+- Do not provide raw user input to `open` methods.
+- Always try to use internal Ruby API (if it exists) instead of running an OS command. Use internal language features instead of invoking commands that can be exploited.
+- If the use of user input is unavoidable, create an allowlist for inputs, such as allowed arguments.
+- Strip everything except alphanumeric characters from an input provided for the command string and arguments.
+
+#### Semgrep rule
+
+<LinkToRegistryRule ruleId="ruby.lang.security.dangerous-exec.dangerous-exec" />
+
+### 1.E. Backticks (``) or %x[command] methods
+
+Backticks ``` `` ``` or `%x[command]` methods enable to execute system commands and return their outputs. Both methods accept string interpolation. As for other methods mentioned in this cheat sheet, when this method is used with user input, it can lead to command injection vulnerability.
```

Comment:
Enable who? This is not a gramatically correct sentence I think.

Suggestion:

Backtics or ... methods allow developers to ...